dandiapi/api/mail.py

```diff
@@ -4,11 +4,16 @@
 
 
 def build_message(subject, message, to, html_message):
-    message = mail.EmailMultiAlternatives(subject, message, FROM_EMAIL, [to.email])
+    if type(to) is not str:
+        to = to.email
```

Comment:
Is this necessary because of constraints beyond our control? It's ok if so, but I think we should start adding mypy type hints to our codebase to help ourselves out in codepaths that take `Union` data types like this.

Reply:
I agree, I think a union type here would be good

Reply:
I did this because I wanted to support sending to both `User` objects and string emails (`dandi@mit.edu`), but I now realize I overcomplicated things here. I will rip this out.